internal/provider/resource_aci_rest_managed.go

```diff
@@ -0,0 +1,518 @@
+		aciRestManagedChild.Rn = child.Rn
+
+		var childClassDetails map[string]interface{}
+
+		for _, childClass := range classChildren {
+			childClassDetails = findChildClass(childClass.(map[string]interface{}), &aciRestManagedChild)
+			if childClassDetails != nil {
+				break
+			}
+		}
+
+		childContent := make(map[string]attr.Value, 0)
+		for contentKey := range child.Content.Elements() {
+			if val, ok := childClassDetails["attributes"].(map[string]interface{})[contentKey]; ok && !child.Content.Elements()[contentKey].IsNull() {
+				childContent[contentKey] = basetypes.NewStringValue(val.(string))
+			} else {
+				childContent[contentKey] = basetypes.NewStringNull()
+			}
+		}
+		aciRestManagedChild.Content, _ = types.MapValue(types.StringType, childContent)
+		foundChildren = append(foundChildren, aciRestManagedChild)
+	}
+
+	if len(foundChildren) > 0 {
+		data.Child, _ = types.SetValueFrom(ctx, data.Child.ElementType(ctx), foundChildren)
+	} else {
+		data.Child = types.SetNull(data.Child.ElementType(ctx))
+	}
+
+}
+
+func findChildClass(childClass map[string]interface{}, child *ChildAciRestManagedResourceModel) map[string]interface{} {
+
+	for className, classDetails := range childClass {
+		classRn := classDetails.(map[string]interface{})["attributes"].(map[string]interface{})["rn"].(string)
+		if child.Rn.ValueString() == classRn {
+			child.ClassName = basetypes.NewStringValue(className)
+			return classDetails.(map[string]interface{})
+		}
+	}
+	return nil
+}
+
+func getAciRestManagedChildPayloads(ctx context.Context, diags *diag.Diagnostics, data *AciRestManagedResourceModel, childPlan, childState []ChildAciRestManagedResourceModel) []interface{} {
+	childPayloads := []interface{}{}
+	if !data.Child.IsUnknown() && !data.Child.IsNull() {
+		childIdentifiers := []AciRestManagedChildIdentifier{}
+		for _, child := range childPlan {
+			childMap := map[string]map[string]string{"attributes": {}}
+			if !child.Rn.IsUnknown() {
+				childMap["attributes"]["rn"] = child.Rn.ValueString()
+			}
+			if !child.Content.IsNull() && !child.Content.IsUnknown() {
+				for k, v := range child.Content.Elements() {
+					if !v.(basetypes.StringValue).IsNull() && !v.(basetypes.StringValue).IsUnknown() {
+						childMap["attributes"][k] = v.(basetypes.StringValue).ValueString()
+					}
+				}
+			}
+			childPayloads = append(childPayloads, map[string]interface{}{child.ClassName.ValueString(): childMap})
+			childIdentifier := AciRestManagedChildIdentifier{}
+			childIdentifier.Rn = child.Rn
+			childIdentifier.ClassName = child.ClassName
+			childIdentifiers = append(childIdentifiers, childIdentifier)
+		}
+		for _, child := range childState {
+			delete := true
+			for _, childIdentifier := range childIdentifiers {
+				if childIdentifier.Rn == child.Rn && childIdentifier.ClassName == child.ClassName {
+					delete = false
+					break
+				}
+			}
+			if delete {
+				// TODO check if this works
```

Comment:
Can this TODO be removed now? I think changing the class or rn of child works as expected?
Pretty sure I added this here but may not have removed it.
```suggestion
```

Reply:
removed